### PR TITLE
Add RenameBranch support

### PR DESCRIPTION
**Pull request by vlsokol**
Add support for rename a branch: https://docs.github.com/en/rest/reference/repos#rename-a-branch
Fixes: https://github.com/google/go-github/issues/2118

### CODE REVIEW

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:1051]: I'm thinking of simplifying the API a bit.

How about this:

```suggestion
func (s *RepositoriesService) RenameBranch(ctx context.Context, owner, repo, branch, newName string) (*Branch, *Response, error) {
```

and you can keep your new struct, but you no longer would export it, like this:

```
type renameBranchRequest struct {
	NewName string `json:"new_name"`
}
```

and then just use it within the method by creating a new one on demand:

```
r := &renameBranchRequest{NewName: newName}
req, err := s.client.NewRequest("POST", u, r)
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @vlsokol !
Let me know what you think about simplifying the API endpoint:

File: github/repos.go
Comment (vlsokol) [reply github/repos.go]: @gmlewis, thank you.
Please take a look at https://github.com/google/go-github/pull/2119/commits/c9e1ab682b7241b975aa7f3a0361592ab57e7660

**Review by gmlewis — APPROVED** [review]
This looks great, @vlsokol ... thank you!
LGTM.

Awaiting second LGTM (from any repo contributor) before merging.

**Review by AGMETEOR — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @AGMETEOR !
Merging.

**Pull request merged**
